lib/demo_web/templates/layout/_user_menu.html.eex

```diff
@@ -0,0 +1,10 @@
+<ul>
+<%= if @current_user do %>
+  <li><%= @current_user.email %></li>
+  <li><%= link "Settings", to: Routes.user_settings_path(@conn, :edit) %></li>
+  <li><%= link "Logout", to: Routes.user_session_path(@conn, :delete), method: :delete %></li>
+<% else %>
+  <li><%= link "Register", to: Routes.user_registration_path(@conn, :new) %></li>
+  <li><%= link "Login", to: Routes.user_session_path(@conn, :new) %></li>
+<% end %>
+</ul>
```

Comment:
We will generate the menu as a separate file. We can either inject it automatically in the root (or app) layout right after the opening "<body>".